apps/ledger-live-mobile/src/screens/Settings/Debug/Generators/index.tsx

```diff
@@ -1,15 +1,88 @@
-import React from "react";
+import React, { useCallback } from "react";
 import config from "react-native-config";
 import { getEnv } from "@ledgerhq/live-common/env";
+import { Alert as Confirmation } from "react-native";
+import { Alert, Icons } from "@ledgerhq/native-ui";
+import { useDispatch, useSelector } from "react-redux";
+import { State } from "../../../../reducers/types";
 import GenerateMockAccounts from "./GenerateMockAccounts";
 import GenerateMockAccountsNft from "./GenerateMockAccountsNFTs";
 import ImportBridgeStreamData from "./ImportBridgeStreamData";
 import GenerateMockAccount from "./GenerateMockAccountsSelect";
 import GenerateAnnouncement from "./GenerateAnnouncementMockData";
 import SettingsNavigationScrollView from "../../SettingsNavigationScrollView";
 import ToggleServiceStatusIncident from "./ToggleServiceStatus";
+import SettingsRow from "../../../../components/SettingsRow";
+import { dangerouslyOverrideState } from "../../../../actions/settings";
+import { useReboot } from "../../../../context/Reboot";
+
+import { INITIAL_STATE as INITIAL_SETTINGS_STATE } from "../../../../reducers/settings";
+import { INITIAL_STATE as INITIAL_ACCOUNTS_STATE } from "../../../../reducers/accounts";
+import { INITIAL_STATE as INITIAL_BLE_STATE } from "../../../../reducers/ble";
 
 export default function Generators() {
+  const state = useSelector<State, State>(s => s);
+  const dispatch = useDispatch();
+  const reboot = useReboot();
+
+  const onCallbackWithConfirmation = (callback: () => void) => {
+    Confirmation.alert(
+      "Destructive operation",
+      "There is no coming back from this.",
+      [
+        {
+          text: "Destroy",
+          onPress: () => callback(),
+        },
+        {
+          text: "Cancel",
+          // eslint-disable-next-line @typescript-eslint/no-empty-function
+          onPress: () => {},
+        },
+      ],
+      {
+        cancelable: true,
+      },
+    );
+  };
+
+  const onWipeSettings = useCallback(() => {
+    onCallbackWithConfirmation(() => {
+      dispatch(
+        dangerouslyOverrideState({
+          ...state,
```

Comment:
do you actually have to pass the current state here? 🤔
the reducers of this action already do that in a safer way

Reply:
Honestly I'm not sure, it's `dangerouslyOverride` why do it safer?

Reply:
but you are probably right, passing a partial state should be enough, in the debug state screen I passed the whole thing to not have to keep track of what I modified, but here it could be just passing in the settings/accounts/ble and it _should_ work

Reply:
I think this is just a bad practice with redux, especially if the reducers already handle this. What I meant by "safer" is in terms of synchronisation with the current state of the store. The current state of the store should be handle at the reducer level to avoid race conditions.
Changing this will also be better in terms of perf because right now with the `useSelector(s => s)` this screen is getting unnecessarily re-rendered at every single dispatch of an action that modifies the store.

Reply:
Good point, I'll change it 👍🏼